ibis/expr/operations.py

```diff
@@ -804,6 +872,38 @@ class Variance(VarianceBase):
     pass
 
 
+class Correlation(Reduction):
+    """
+    coefficient of correlation of a set of number pairs.
+
+    """
+    left = Arg(rlz.numeric)
+    right = Arg(rlz.numeric)
+    how = Arg(rlz.isin({'sample', 'pop'}), default=None)
+    where = Arg(rlz.boolean, default=None)
+
+    def output_type(self):
+        if isinstance(self.left, ir.DecimalValue):
+            dtype = self.left.type().largest
+        else:
+            dtype = dt.float64
+        return dtype.scalar_type()
+
+
+class Covariance(Reduction):
+    left = Arg(rlz.column(rlz.numeric))
+    right = Arg(rlz.column(rlz.numeric))
+    how = Arg(rlz.isin({'sample', 'pop'}), default=None)
+    where = Arg(rlz.boolean, default=None)
+
+    def output_type(self):
```

Comment:
@cpcloud The same as above.

Reply:
Yes @xmnlab leave out DecimalValue stuff for now